velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -482,6 +482,36 @@ struct DateFromUnixDateFunction {
   }
 };
 
+template <typename T>
+struct DateTruncFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  const tz::TimeZone* timeZone_ = nullptr;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* format,
+      const arg_type<Timestamp>* /*timestamp*/) {
+    timeZone_ = getTimeZoneFromConfig(config);
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Timestamp>& result,
+      const arg_type<Varchar>& format,
+      const arg_type<Timestamp>& timestamp) {
+    std::optional<DateTimeUnit> unitOption =
+        fromDateTimeUnitString(format, false /*throwIfInvalid*/, true, true);
+    // Return null if unit is illegal.
+    if (!unitOption.has_value()) {
+      return false;
+    }
+    DateTimeUnit unit = unitOption.value();
```

Comment:
const DateTimeUnit unit. I would prefer to drop this initialization since it is used only once.